src/query-parser/utils/context-factory.ts

```
@@ -0,0 +1,420 @@
+
+    const baseContext = contexts[0];
+    if (!baseContext) {
+      throw new Error("Base context cannot be undefined");
+    }
+
+    const mergedContext = contexts.reduce((merged, current) => {
+      if (!current) {
+        return merged;
+      }
+      return {
+        ...merged,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)